ibis/expr/signature.py

```diff
@@ -81,17 +102,22 @@ def Argument(validator, default=EMPTY):
         If callable is passed, it will be executed just before the inner,
         and itsreturn value will be treaded as default.
     """
+    warnings.warn(
+        "ibis.expr.signature.Argument will be removed in ibis 3.0",
```

Comment:
Should be changed to 4.0.0 right? The idea is to keep it around for one more major release and then remove it.

Also, there's a new `ibis.util.deprecated` decorator to use here.